packages/components/socialFeed/Map/Map.web.tsx

```diff
@@ -134,9 +134,11 @@ export const Map: FC<MapProps> = ({
   const markers: MarkerPopup[] = useMemo(() => {
     if (!posts) return [];
     const results: MarkerPopup[] = [];
-    posts.forEach((post, index) => {
+    posts.forEach((post) => {
       const metadata = zodTryParseJSON(
-        ZodSocialFeedPostMetadata,
+        ZodSocialFeedPostMetadata.partial().merge(
+          ZodSocialFeedPostMetadata.pick({ title: true }),
```

Comment:
You don't need to modify the Zod type to try parse.

The issue is due to the fact we have 4 differents types for the fetched posts.
Here we only try to parse `ZodSocialFeedPostMetadata`. So, the tracks and videos posts validation fails.

The `result` here has no `success`: https://github.com/TERITORI/teritori-dapp/blob/bugfix/video-audio-map/packages/utils/sanitize.ts#L22
If you `console.log` the `result`, you will see what's wrong to validate `ZodSocialFeedPostMetadata`

To avoid that, because the map gathers all posts categories, we must handle these Zod types. Here is my suggestion:

```
const videoPostMetadata = zodTryParseJSON(
  ZodSocialFeedVideoMetadata,
  post.metadata,
);
const trackMetadata = zodTryParseJSON(
  ZodSocialFeedTrackMetadata,
  post.metadata,
);
const articleMetadata = zodTryParseJSON(
  ZodSocialFeedArticleMetadata,
  post.metadata,
);
const postMetadata = zodTryParseJSON(
  ZodSocialFeedPostMetadata,
  post.metadata,
);
const metadataToUse = videoPostMetadata || trackMetadata || articleMetadata || postMetadata;
```

We need a refacto about it. Either unify all types in one, either add an utilitary function to handle (try parse) these differents zod types

Reply:
actually no there is the `zodSocialFeedCommonMetadata` when you need to only parse the title

Reply:
I have made a fix for this discussion in this commit: https://github.com/TERITORI/teritori-dapp/pull/1400/commits/3a16113c647d667b39245f99481859b42872c99b

Reply:
> actually no there is the `zodSocialFeedCommonMetadata` schema when you need to only parse the title

In this case, there are other things that i need to parse also based on the category type, so I have created a util function to handle those cases.

Reply:
you can switch on the post type to do the proper parsing in each case, the whole point of using zod is to have type safety, using a `z.ZodTypeAny` entirely defeat this purpose